examples/time_series/goes_xrs_nrt_data.py

```diff
@@ -64,9 +69,30 @@
 goes_ts = ts.TimeSeries(goes_data, meta, units, source="xrs")
 
 ###############################################################################
-# Finally, we can plot the timeseries.
+# NOAA also provides the past 7 days of flare event identifications which we 
+# can also parse.
+# Note that it is possible that no flares occurred in the last 7 days.
+
+flare_events = pd.read_json(
+    "https://services.swpc.noaa.gov/json/goes/primary/xray-flares-7-day.json"
+)
+
+###############################################################################
+# Next we sort the list and select only the largest flares. 
+
+if len(flare_events) > 0:
+    largest_flares = flare_events.sort_values('max_xrlong', ascending=False)[0:5]
+else:
+    largest_flares = []
+###############################################################################
+# Finally, we can plot the timeseries and overlay all the flares that occurred.
+# We will do this only for the last day otherwise the plot may be too busy.
 
 fig, ax = plt.subplots()
 goes_ts.plot(axes=ax)
-
+for index, this_flare in largest_flares.iterrows():
+    peak_time = parse_time(this_flare["max_time"])
+    ax.axvline(peak_time.datetime, label=f'{peak_time.datetime} {this_flare["max_class"]}')
```

Comment:
```suggestion
if len(largest_flares) > 0:
    for _, this_flare in largest_flares.iterrows():
        peak_time = parse_time(this_flare["max_time"])
        ax.axvline(peak_time.datetime, label=f'{peak_time.datetime} {this_flare["max_class"]}')
```
We can't call `.iterrows()` on an empty list.  There's probably a nicer way to write this, but this fix is acceptable enough to merge.